src/VisualStudio/IntegrationTest/IntegrationTests/AbstractEditorTest.cs

```diff
@@ -65,6 +72,60 @@ protected void SetUpEditor(string markupCode)
         protected void AddFile(string fileName, string contents = null, bool open = false)
             => VisualStudio.Instance.SolutionExplorer.AddFile(ProjectName, fileName, contents, open);
 
+        protected void OpenFile(string projectName, string fileName)
+            => VisualStudio.Instance.SolutionExplorer.OpenFile(projectName, fileName);
+
+        protected void OpenFileWithDesigner(string projectName, string fileName)
+            => VisualStudio.Instance.SolutionExplorer.OpenFileWithDesigner(projectName, fileName);
+
+        protected void CloseFile(string projectName, string fileName, bool saveFile = true)
+            => VisualStudio.Instance.SolutionExplorer.CloseFile(projectName, fileName, saveFile);
+
+        protected void SaveFile(string projectName, string fileName)
+            => VisualStudio.Instance.SolutionExplorer.SaveFile(projectName, fileName);
+
+        protected void AddWinFormButton(string projectName, string fileName, string buttonName)
+        {
+            OpenFileWithDesigner(projectName, fileName);
+            VisualStudio.Instance.Editor.AddWinFormButton(buttonName);
+        }
+
+        protected void DeleteWinFormButton(string projectName, string fileName, string buttonName)
+        {
+            OpenFileWithDesigner(projectName, fileName);
+            VisualStudio.Instance.Editor.DeleteWinFormButton(buttonName);
```

Comment:
It seems strange to me that so many of these methods call `OpenFileWithDesigner` first. In many cases we're going to be executing that action unnecessarily (e.g. when you have `AddWinFormButton` and `EditWinFormButtonProperty` calls back to back). I'd rather the call to `OpenFileWithDesigner` just be pulled out into the test itself. That would also make it clear that you're interacting with the existing, open document.